logger/logger.go

```diff
@@ -64,6 +64,10 @@ type logger struct {
 	Logger kitlog.Logger
 }
 
+func (l *logger) Decorate(keyvals ...interface{}) {
+	l.Logger = kitlog.NewContext(l.Logger).With(keyvals...)
```

Comment:
I do not see any real benefit out of this here. I will unsubscribe now and you guys can do what you think is right. I do not think we need this, but I am maybe wrong. My two cents.